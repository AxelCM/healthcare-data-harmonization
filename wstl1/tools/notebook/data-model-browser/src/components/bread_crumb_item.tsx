/**
 * @jsx React.createElement
 * @jsxFrag React.Fragment
 */
// Copyright 2020 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// tslint:disable-next-line:enforce-name-casing
import * as React from 'react';
import Link from '@material-ui/core/Link';

/**
 * Properties for BreadCrumbItem component.
 */
interface BreadCrumbItemProps {
  /**
   * The JSON path components leading to a leaf bread crumb item.
   */
  path: string[];

  /**
   * The display label for the bread crumb item.
   */
  label: string;

  /**
   * The callback function invoked when the bread crumb item is clicked. The
   * callback argument are the path components of the bread crumb item.
   */
  onClick: (path: string[]) => void;
}

/**
 * Component that represents a single bread crumb item.
 */
export class BreadCrumbItem extends React.PureComponent<BreadCrumbItemProps> {
  /**
   * Constructs a new BreadCrumbItem.
   */
  constructor(props: BreadCrumbItemProps) {
    super(props);
  }

  /**
   * Render the BreadCrumbItem component.
   */
  render() {
    return (
      <Link
        color="textSecondary"
        href={
          this.props.path.length > 0
            ? this.props.path[this.props.path.length - 1]
            : '#'
        }
        onClick={(e: React.MouseEvent<HTMLAnchorElement>) => {
          e.preventDefault();
          this.props.onClick(this.props.path);
        }}
      >
        {this.props.label}
      </Link>
    );
  }
}
