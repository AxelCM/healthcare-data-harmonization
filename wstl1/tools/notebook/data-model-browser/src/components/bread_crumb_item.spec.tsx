/**
 * @jsx React.createElement
 * @jsxFrag React.Fragment
 */
// Copyright 2020 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// tslint:disable-next-line:enforce-name-casing
import * as React from 'react';
import {shallow} from 'enzyme';

import Link from '@material-ui/core/Link';
import {BreadCrumbItem} from './bread_crumb_item';

describe('BreadCrumbItem', () => {
  const click = jest.fn();
  const fakeField = 'fake_field';
  const path = ['FHIR', 'FakeResource', fakeField];
  const wrapper = shallow(
    <BreadCrumbItem
      path={path}
      label={fakeField}
      onClick={click}
    />
  );

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render children', () => {
    expect(wrapper.contains(fakeField)).toBeTruthy();
  });

  it('should call the correct function on click', () => {
    wrapper.find(Link).simulate('click', {
      preventDefault: () => {},
    });
    expect(click).toHaveBeenCalled();
  });
});
