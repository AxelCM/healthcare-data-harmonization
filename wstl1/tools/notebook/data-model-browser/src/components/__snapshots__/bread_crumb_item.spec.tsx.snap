// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbItem should render correctly 1`] = `
<WithStyles(ForwardRef(Link))
  color="textSecondary"
  href="fake_field"
  onClick={[Function]}
>
  fake_field
</WithStyles(ForwardRef(Link))>
`;
